### PR TITLE
Nova quotas, dump user passwords

### PR DESCRIPTION
**Pull request by toha10**
Implemented nova quotas migration via Nova API.
Added dump user passwords to csv file if we generate new ones.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/101/

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/102/

**Comment by toha10** [comment]
BTW guys, how do you think: do we still need 'deprecated' methods for working with nova quotas via DB ? or we can complete delete these methods (_read_info_quotas, _deploy_user_quotas, _deploy_project_quotas) ?

**Comment by miarmak** [comment]
Looks good to me.
 
And what about quotas via DB: I like API way and if we do not have any reasonable reason to migrate quotas via database - let's get rid of it) As I remember, we don't.

**Comment by MirantisWorkloadMobility** [comment]
I don't remember any special requirements for working with quotas via DB
too.

On Wed, Mar 11, 2015 at 8:24 PM, Maksym Iarmak <notifications@github.com>
wrote:

> Looks good to me.
>
> And what about quotas via DB: I like API way and if we do not have any
> reasonable reason to migrate quotas via database - let's get rid of it) As
> I remember, we don't.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MirantisWorkloadMobility/CloudFerry/pull/111#issuecomment-78311641>
> .
>



-- 
Anton Samoylov,
Deployment Engineer, Mirantis, Inc.
asamoylov@mirantis.com

**Comment by mirrorcoder** [comment]
+1

File: cloudferrylib/utils/utils.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import ipaddr
 import yaml
 from logging import config
+import csv
```
Comment (roman-verchikov) [inline cloudferrylib/utils/utils.py]: Please add this dependency to requirements.txt

File: cloudferrylib/os/compute/nova_compute.py [hunk 2]
```diff
@@ -93,10 +95,32 @@ def _read_info_resources(self, **kwargs):
             info['flavors'][flavor.id] = self.convert(flavor)
 
         if self.config.migrate.migrate_quotas:
-            self._read_info_quotas(info)
```
Comment (roman-verchikov) [inline cloudferrylib/os/compute/nova_compute.py]: Please remove the implementation as well, it's not used anymore

File: cloudferrylib/os/identity/keystone.py [hunk 3]
```diff
@@ -377,3 +387,7 @@ def _render_template(self, name_file, args):
             return self.templater.render(name_file, args)
         else:
             return None
+
+    def dump_passes_to_file(self, info, file_path="pass_list.csv"):
+        table_header = ['Tenant', 'User', 'Password']
+        write_csv(info=info, csv_file_path=file_path, header=table_header)
```
Comment (roman-verchikov) [inline cloudferrylib/os/identity/keystone.py]: New line missing

File: cloudferrylib/os/identity/keystone.py [hunk 4]
```diff
@@ -286,12 +290,18 @@ def _deploy_users(self, users, tenants):
             _user['meta']['new_id'] = self.create_user(user['name'], password,
                                                        user['email'],
                                                        tenant_id).id
-            if self.config['migrate']['keep_user_passwords']:
+
+            if keep_passwd:
                 _user['meta']['overwrite_password'] = True
```
Comment (roman-verchikov) [inline cloudferrylib/os/identity/keystone.py]: This looks counter-intuitive to me. I read this as "If CloudFerry is configured to keep old user passwords, then modify user to have password overwritten (some time later)". Not sure whether it's me or the code wrong.

**Comment by roman-verchikov** [comment]
1. Please fix inline comments;
2. Please squash 5 commits into one;
3. Consider covering implemented functionality with unit tests (although given how the code is structured it may get very tricky)

**Comment by roman-verchikov** [comment]
This needs to be rebased in addition to comments above

**Pull request closed without merging**